dan_layer/core/src/workers/consensus_worker.rs

```diff
@@ -406,42 +407,84 @@ mod test {
         models::{
             Committee,
             ConsensusWorkerState::{Commit, Decide, NextView, PreCommit, Prepare},
+            TariDanPayload,
         },
         services::{
             infrastructure_services::mocks::{mock_outbound, MockInboundConnectionService, MockOutboundService},
-            mocks::{mock_events_publisher, MockCommitteeManager, MockEventsPublisher},
+            mocks::{
+                mock_base_node_client,
+                mock_checkpoint_manager,
+                mock_events_publisher,
+                mock_payload_processor,
+                mock_signing_service,
+                mock_static_payload_provider,
+                MockChainStorageService,
+                MockCommitteeManager,
+                MockEventsPublisher,
+                MockServiceSpecification,
+                MockValidatorNodeClientFactory,
+            },
         },
+        storage::mocks::MockDbFactory,
     };
 
     fn start_replica(
-        _inbound: MockInboundConnectionService<&'static str, &'static str>,
-        _outbound: MockOutboundService<&'static str, &'static str>,
-        _committee_manager: MockCommitteeManager,
-        _node_id: &'static str,
-        _shutdown_signal: ShutdownSignal,
-        _events_publisher: MockEventsPublisher<ConsensusWorkerDomainEvent>,
+        inbound: MockInboundConnectionService<RistrettoPublicKey, TariDanPayload>,
```

Comment:
I intentionally tried to use strings here instead of the more complicated structs, if they can't be used, it means there is a type that uses a PublicKey instead of a generic somewhere

Reply:
The root cause of needing PublicKey is the ConsensusWorker specification:

`impl<TSpecification: ServiceSpecification<Addr = PublicKey>> ConsensusWorker<TSpecification> {`